cluster-autoscaler/core/podlistprocessor/filter_out_expendable.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podlistprocessor
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	core_utils "k8s.io/autoscaler/cluster-autoscaler/core/utils"
+	caerrors "k8s.io/autoscaler/cluster-autoscaler/utils/errors"
+	klog "k8s.io/klog/v2"
+)
+
+type filterOutExpendable struct {
+}
+
+// NewFilterOutExpendablePodListProcessor creates a PodListProcessor filtering out expendable pods
+func NewFilterOutExpendablePodListProcessor() *filterOutExpendable {
+	return &filterOutExpendable{}
+}
+
+// Process filters out pods which are expendable and adds pods which is waiting for lower priority pods preemption to the cluster snapshot
+func (p *filterOutExpendable) Process(context *context.AutoscalingContext, pods []*apiv1.Pod) ([]*apiv1.Pod, error) {
+	nodes, err := context.AllNodeLister().List()
+	if err != nil {
+		return nil, err
```

Comment:
Looks like static_autoscaler ignores the error here, let's make sure it is handled properly.

Reply:
Added a warning here, if nil is returned to static_autoscaler unschedulable pods would be empty and no scale up would occur so it's implicitly handled. (Same is also being done at `filterOutSchedulablePodListProcessor` without the new warning)

Reply:
I don't think the warning and returning the error at the same time is useful. Errors should either be handled or passed back to the caller, logging & returning tries to do both. Often error handling is responsibility of the caller when function can return `error` type, but I see a problem here - the return value is ignored: https://github.com/kubernetes/autoscaler/blob/dcaba4483d512f4a93b1e60c7272e1de8fe181a9/cluster-autoscaler/core/static_autoscaler.go#L518

Reply:
Added warnings @ static_autoscaler & removed the one in processor. The processor now returns a formatted error